### DISCUSSION TITLE
How to use dynamo's SIZE function in a condition

### DISCUSSION THREAD

**Discussion by alxAgu** [dynamoose/dynamoose#1509] [Q&A]
Hi, I was wondering if there is any way to use the `size` function as part of a condition?

Eg:
```
// Find products with less than 5 colours
const products = ProductModel.query({ productId }).where('colours').size().lt(5);
```
** `colours` is a String SET

**Comment by fishcharlie** [comment]
@alxAgu No idea what you mean here. What does `size` represent? That isn't a valid Dynamoose function.

**Reply by alxAgu** [reply]
@fishcharlie, checkout the last item in the table: https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Expressions.OperatorsAndFunctions.html#Expressions.OperatorsAndFunctions.Functions

**Reply by alxAgu** [reply]
Basically, an equivalent for:
```
const condition = {
  TableName: env.TABLE_NAME,
  KeyConditionExpression: 'productId = :productId',
  FilterExpression: 'size(colours) < :maxColours', // <-- this is what I'm after
  ExpressionAttributeValues: {
    ':productId': { 'S': productId },
    ':maxColours': { 'N': '5' },
  },
};
```

**Reply by fishcharlie** [reply]
@alxAgu Not currently supported. Feel free to submit a feature request.

**Reply by alxAgu** [reply]
I just did: https://github.com/dynamoose/dynamoose/issues/1510

Thanks.
